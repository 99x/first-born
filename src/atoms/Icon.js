import { Platform } from "react-native";
import React, { Component } from "react";
import Ionicon from "react-native-vector-icons/Ionicons";
import PropTypes from "prop-types";
import { commonColors } from "../utils/color";

export class Icon extends Component {
    render() {
        const { name, size, color, ...otherProps } = this.props;

        if (!name.includes("logo")) {
            return (
                <Ionicon
                    name={
                        Platform.OS === "android" ? "md-" + name : "ios-" + name
                    }
                    size={size}
                    color={color}
                    {...otherProps}
                />
            );
        }
        return (
            <Ionicon name={name} size={size} color={color} {...otherProps} />
        );
    }
}

Icon.propTypes = {
    ...Ionicon.propTypes,
    name: PropTypes.string
};

Icon.defaultProps = {
    color: commonColors.white,
    size: 18
};
