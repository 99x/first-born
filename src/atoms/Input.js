import React, { Component } from "react";
import { TextInput, StyleSheet, Platform, Keyboard, View } from "react-native";
import PropTypes from "prop-types";

import { commonColors } from "../utils/color";

export class Input extends Component {

    constructor(props) {
        super(props);
        this.state = {
            focused: false,
            error: false
        }
    }

    onChangeText = (text) => {
        const { isValid, onChangeText } = this.props;
        onChangeText(text);
        if (isValid) {
            this.setState({ error: isValid(text) });
        }
    }

    render() {
        const { onChangeText, placeholder, color, ...otherProps } = this.props;

        const underLineColor = !color ? commonColors.black : color

        if (Platform.OS === "android") {
            return (
                <View style={this.state.focused ? [styles.input, { borderColor: underLineColor, borderWidth: 2 }] : this.state.error ? [styles.input, { borderColor: "#e74c3c", borderWidth: 2 }] : styles.input}>
                    <TextInput
                        ref={(c) => this._root = c}
                        style={{ width: "100%" }}
                        underlineColorAndroid={"transparent"}
                        placeholder={placeholder}
                        placeholderTextColor={this.state.error ? '#e74c3c' : 'rgba(33, 33, 33, 0.5)'}
                        onChangeText={this.onChangeText}
                        onFocus={() => this.setState({ focused: true })}
                        onBlur={() => this.setState({ focused: false })}
                        onSubmitEditing={() => { this.setState({ focused: true }); Keyboard.dismiss() }}
                        {...otherProps}
                    />
                </View>
            )
        }
        return (
            <TextInput
                ref={(c) => this._root = c}
                style={styles.input}
                underlineColorAndroid={"transparent"}
                placeholder={placeholder}
                onChangeText={onChangeText}
                {...otherProps}
            />
        )
    }
}

Input.propTypes = {
    color: PropTypes.string,
    ...TextInput.propTypes
}

const styles = StyleSheet.create({
    input: {
        borderRadius: 10,
        paddingHorizontal: 5,
        marginVertical: 10,
        width: '100%',
        borderColor: "rgba(33, 33, 33, 0.5)",
        borderWidth: 0.9,
        height: 45
    }
})
