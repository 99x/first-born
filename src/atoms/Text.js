import React, { Component } from "react";
import { Text } from "react-native";
import PropTypes from 'prop-types';
import { commonColors } from "../utils/color";
import { getFontSize } from "../variables/textSizeVariables";

class TextAE extends Component {

    render() {
        const { color, size, bold, children, ...otherProps } = this.props;

        const textColor = !color ? commonColors.black : color;

        const textSize = !size ? "p" : size;

        const textBold = !bold ? "normal" : "bold";

        const textStyle = { color: textColor, fontSize: getFontSize(textSize), fontWeight: textBold };

        return (
            <Text
                ref={c => (this._root = c)}
                style={textStyle}
                {...otherProps}
            >
                {children}
            </Text>
        )
    }
}

TextAE.propTypes = {
    bold: PropTypes.bool,
    color: PropTypes.string,
    size: PropTypes.oneOf(["h1", "h2", "h3", "h4", "h5", "h6", "p", "callout", "sub_heading", "footnote", "caption_big", "caption_small"]),
    ...Text.propTypes
};

TextAE.defaultProps = {
    size: "p",
    bold: false,
    color: commonColors.black
};

export { TextAE as Text };
