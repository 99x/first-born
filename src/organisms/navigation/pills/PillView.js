import React, { Component } from "react";
import { View, StyleSheet, ScrollView, Animated } from "react-native";
import PropTypes from "prop-types";

import { PillBar } from "./PillBar";

export class PillView extends Component {
    constructor(props) {
        super(props);

        this.state = {
            activePill: 0
        };

        this.fadeAnim = new Animated.Value(1);
    }

    animate() {
        this.fadeAnim = new Animated.Value(0);
        Animated.timing(this.fadeAnim, {
            toValue: 1,
            duration: 250,
            useNativeDriver: true
        }).start();
    }

    handlePillChange = activePill => {
        this.setState({ activePill: activePill });
        this.animate();
    };

    render() {
        const { pillHeaders, color, ...otherProps } = this.props;

        return (
            <View style={styles.container}>
                <PillBar
                    pillHeaders={pillHeaders}
                    color={color}
                    onPillChange={this.handlePillChange}
                    activePill={this.state.activePill}
                />
                <ScrollView>
                    <Animated.View style={{ opacity: this.fadeAnim }}>
                        {this.renderScene()}
                    </Animated.View>
                </ScrollView>
            </View>
        );
    }

    renderScene() {
        return this.props.pillScenes[this.state.activePill].scene;
    }
}

PillView.propTypes = {
    pillScenes: PropTypes.array.isRequired,
    pillHeaders: PropTypes.array.isRequired,
    color: PropTypes.string,
    ...View.propTypes
};

const styles = StyleSheet.create({
    container: {
        flex: 1
    }
});
