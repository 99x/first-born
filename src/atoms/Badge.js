import React, { Component } from "react";
import { View, StyleSheet, Platform, Image } from "react-native";
import PropTypes from "prop-types";
import { Icon } from "./Icon";
import { commonColors } from "../utils/color";
import { Text } from "./Text";

export class Badge extends Component {
    render() {
        const {
            color,
            children,
            ...otherProps
        } = this.props;

        const badgeColor = color;

        let badgeStyle = [styles.defaultBadge, {
            borderRadius: 15,
            backgroundColor: badgeColor
        }];

        let textStyle = {
            color: commonColors.white
        };

        let iconColor = commonColors.white;;

        const newChildren = React.Children.map(children, child =>
            child && child.type === Text
                ? React.cloneElement(child, {
                    ...child.props,
                    style: { ...child.props.style, ...textStyle },
                    size: "footnote"
                })
                : child && child.type === Icon
                    ? React.cloneElement(child, {
                        ...child.props,
                        color: iconColor,
                        size: 15
                    })
                    : child && child.type === Image
                        ? React.cloneElement(child, {
                            ...child.props,
                            style: {
                                ...child.props.style,
                                width: 15,
                                height: 15
                            }
                        })
                        : null
        );

        return (
            <View
                ref={c => (this._root = c)}
                style={badgeStyle}
                {...otherProps}
            >
                {newChildren}
            </View>
        );
    }
}

Badge.propTypes = {
    color: PropTypes.string,
    ...View.propTypes
};

Badge.defaultProps = {
    color: commonColors.primary,
};

const styles = StyleSheet.create({
    defaultBadge: {
        margin: 5,
        flexDirection: "row",
        justifyContent: "center",
        alignItems: "center",
        paddingHorizontal: 10,
        paddingVertical: 5
    },
});
