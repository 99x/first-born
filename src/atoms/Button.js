import React, { Component } from "react";
import { TouchableOpacity, StyleSheet, Platform, Image } from "react-native";
import PropTypes from "prop-types";
import { Icon } from "./Icon";
import { getFontSize, getButtonPadding, getRoundRadius, getIconSize } from "../variables/buttonSizeVariables";
import { commonColors } from "../utils/color";
import { Text } from "./Text";

export class Button extends Component {
    render() {
        const { title, color, rounded, outline, block, size, secondary, transparent, ...otherProps } = this.props;

        const buttonColor = secondary ? commonColors.secondary : color;

        let buttonStyle = [styles.defaultButton];
        let iconSize = getIconSize(size);
        let textStyle;
        let iconColor;

        if (block) {
            buttonStyle.push(styles.blockButton);
        }

        if (rounded) {
            buttonStyle.push({ paddingHorizontal: getButtonPadding(size) + (getButtonPadding(size) / 3), borderRadius: getRoundRadius(size) });
        }

        if (outline) {
            textStyle = { color: buttonColor, fontSize: getFontSize(size) };
            buttonStyle.push({ borderColor: buttonColor, borderWidth: 1, padding: getButtonPadding(size) });
            iconColor = buttonColor;
        } else {
            textStyle = { color: commonColors.white, fontSize: getFontSize(size) };
            buttonStyle.push({ backgroundColor: buttonColor, padding: getButtonPadding(size) });
            iconColor = commonColors.white;
        }

        if (transparent) {
            buttonStyle.push(styles.transparent);
        }

        const children = React.Children.map(this.props.children, child => child && child.type === Text ?
            React.cloneElement(child, { style: textStyle, ...child.props }) : child && child.type === Icon ?
                React.cloneElement(child, { color: iconColor, size: iconSize, ...child.props }) : child && child.type === Image ?
                    React.cloneElement(child, { style: { width: 25, height: 25 }, ...child.props }) : null);

        return (
            <TouchableOpacity ref={(c) => this._root = c} style={buttonStyle} {...otherProps}>
                {children}
            </TouchableOpacity>
        )
    }
}

Button.propTypes = {
    rounded: PropTypes.bool,
    outline: PropTypes.bool,
    block: PropTypes.bool,
    title: PropTypes.string,
    color: PropTypes.string,
    transparent: PropTypes.bool,
    size: PropTypes.oneOf(["small", "default", "large"]),
    ...TouchableOpacity.propTypes
}

Button.defaultProps = {
    color: commonColors.primary,
    size: "default"
}

const styles = StyleSheet.create({
    text: {
        color: commonColors.white
    },
    defaultButton: {
        ...Platform.select({
            android: {
                borderRadius: 5,
                elevation: 3,
            },
            ios: {
                borderRadius: 10,
                shadowOffset: { width: 3, height: 3 },
                shadowOpacity: 0.3,
                shadowRadius: 3,
            }
        }),
        margin: 10,
        shadowColor: "#000",
    },
    blockButton: {
        width: "100%",
        justifyContent: "center",
        alignItems: "center",
    },
    transparent: {
        borderWidth: 0,
        backgroundColor: "transparent",
        borderRadius: 0,
        shadowOffset: { width: 0, height: 0 },
        shadowOpacity: 0,
        shadowRadius: 0,
        elevation: 0,
    }
})
