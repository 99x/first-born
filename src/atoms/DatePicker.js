import React, { Component } from "react";
import {
    TextInput,
    StyleSheet,
    View,
    Modal,
    Platform,
    DatePickerIOS,
    DatePickerAndroid,
    TimePickerAndroid
} from "react-native";
import PropTypes from "prop-types";
import { Icon } from "./Icon";
import { commonColors } from "../utils/color";
import { Text } from "./Text";

export class DatePicker extends Component {
    constructor(props) {
        super(props);
        this.state = {
            focused: false,
            defaultDate: props.defaultDate ? props.defaultDate : new Date(),
            chosenDate:
                !props.placeholder && props.defaultDate
                    ? props.defaultDate
                    : undefined
        };
    }

    render() {
        const {
            placeholder,
            color,
            modalTransparent,
            animationType,
            minimumDate,
            maximumDate,
            locale,
            timeZoneOffsetInMinutes,
            mode
        } = this.props;

        const { focused, chosenDate, defaultDate } = this.state;

        if (Platform.OS === "android") {
            return (
                <View
                    style={
                        focused
                            ? [
                                styles.datePickerAndroid,
                                { borderColor: color, borderWidth: 2 }
                            ]
                            : styles.datePickerAndroid
                    }
                >
                    <Text
                        ref={c => (this._root = c)}
                        style={!chosenDate ? styles.input : styles.inputValue}
                        onPress={() => mode === "date" ? this.openAndroidDatePicker() : this.openAndroidTimePicker()}
                    >
                        {chosenDate
                            ? this.formatChosenDate(chosenDate)
                            : placeholder}
                    </Text>
                    <Icon
                        name="calendar"
                        onPress={() => mode === "date" ? this.openAndroidDatePicker() : this.openAndroidTimePicker()}
                        style={styles.icon}
                        color={focused ? color : commonColors.inputGrey}
                        size={20}
                    />
                </View>
            );
        }
        return (
            <View>
                <View
                    style={
                        focused
                            ? [
                                styles.datePickerIos,
                                { borderColor: color, borderWidth: 2 }
                            ]
                            : styles.datePickerIos
                    }
                >
                    <Text
                        ref={c => (this._root = c)}
                        style={!chosenDate ? styles.input : styles.inputValue}
                        onPress={() => this.openIosDatePicker()}
                    >
                        {chosenDate
                            ? this.formatChosenDate(chosenDate)
                            : placeholder}
                    </Text>
                    <Icon
                        onPress={() => this.openIosDatePicker()}
                        name="calendar"
                        style={styles.icon}
                        color={focused ? color : commonColors.inputGrey}
                        size={20}
                    />
                </View>
                <Modal
                    supportedOrientations={["portrait", "landscape"]}
                    animationType={animationType}
                    transparent={modalTransparent}
                    visible={focused}
                    onRequestClose={() => { }}
                >
                    <Text
                        onPress={() => this.setState({ focused: false })}
                        style={{ flex: 1 }}
                    />
                    <DatePickerIOS
                        date={chosenDate ? chosenDate : defaultDate}
                        onDateChange={this.setDate.bind(this)}
                        minimumDate={minimumDate}
                        maximumDate={maximumDate}
                        mode={mode}
                        locale={locale}
                        timeZoneOffsetInMinutes={timeZoneOffsetInMinutes}
                        style={{ flex: 1, backgroundColor: commonColors.white }}
                    />
                    <Text
                        onPress={() => this.setState({ focused: false })}
                        style={{ flex: 1 }}
                    />
                </Modal>
            </View>
        );
    }

    setDate(date) {
        this.setState({ chosenDate: new Date(date) });
        if (this.props.onDateChange) {
            this.props.onDateChange(date);
        }
    }

    openIosDatePicker() {
        this.setState({ focused: true });
    }

    async openAndroidTimePicker() {
        try {
            this.setState({ focused: true });
            const pickedTime = await TimePickerAndroid.open({
                is24Hour: this.props.is24Hour,
                mode: "default"
            });

            const { action, hour, minute } = pickedTime;

            if (action !== TimePickerAndroid.dismissedAction) {
                const currentDate = new Date();
                const selectedDate = new Date(currentDate.getFullYear(), currentDate.getMonth(), currentDate.getDay(), hour, minute);
                this.setDate(selectedDate);
            }
            this.setState({ focused: false });
        } catch ({ code, message }) {
            console.warn('Cannot open time picker', message);
            this.setState({ focused: false });
        }
    }

    async openAndroidDatePicker() {
        try {
            this.setState({ focused: true });
            const pickedDate = await DatePickerAndroid.open({
                date: this.state.chosenDate
                    ? this.state.chosenDate
                    : this.state.defaultDate,
                minDate: this.props.minimumDate,
                maxDate: this.props.maximumDate,
                mode: "default"
            });

            const { action, year, month, day } = pickedDate;

            if (action === DatePickerAndroid.dateSetAction) {
                const selectedDate = new Date(year, month, day);
                this.setDate(selectedDate);
            }
            this.setState({ focused: false });
        } catch ({ code, message }) {
            console.warn("Cannot open date picker", message);
            this.setState({ focused: false });
        }
    }

    formatChosenDate(date) {

        if (this.props.formatChosenDate) {
            return this.props.formatChosenDate(date);
        }
        if (this.props.mode === "date") {
            return [date.getDate(), date.getMonth() + 1, date.getFullYear()].join(
                "/"
            );
        } else {
            return [date.getHours(), date.getMinutes()].join(
                ":"
            );
        }
    }
}

DatePicker.propTypes = {
    color: PropTypes.string,
    formatChosenDate: PropTypes.func,
    onDateChange: PropTypes.func,
    defaultDate: PropTypes.object,
    minimumDate: PropTypes.object,
    maximumDate: PropTypes.object,
    locale: PropTypes.string,
    is24Hour: PropTypes.bool,
    modalTransparent: PropTypes.bool,
    animationType: PropTypes.string,
    mode: PropTypes.oneOf(["date", "time"]),
    ...TextInput.propTypes
};

DatePicker.defaultProps = {
    modalTransparent: true,
    color: commonColors.primary,
    animationType: "fade",
    placeholder: "Select Date",
    mode: "date",
    is24Hour: true
};

const styles = StyleSheet.create({
    datePickerAndroid: {
        width: "100%",
        flexDirection: "row",
        marginVertical: 10,
        justifyContent: "center",
        alignItems: "center",
        height: 45,
        paddingHorizontal: 10,
        borderRadius: 10,
        borderColor: commonColors.inputGrey,
        borderWidth: 0.9
    },
    datePickerIos: {
        width: "100%",
        flexDirection: "row",
        borderRadius: 15,
        marginVertical: 10,
        borderColor: commonColors.inputGrey,
        borderWidth: 0.9,
        justifyContent: "center",
        alignItems: "center",
        height: 45,
        paddingLeft: 10
    },
    icon: {
        paddingRight: 15
    },
    input: {
        flex: 1,
        color: commonColors.inputGrey
    },
    inputValue: {
        flex: 1,
        color: commonColors.black
    }
});
