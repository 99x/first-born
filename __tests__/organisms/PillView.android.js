import "react-native";
import { View } from "react-native";
import React from "react";
import renderer from "react-test-renderer";
import { Text } from "../../src/atoms/Text";
import { PillView } from "../../src/organisms/navigation/pills/PillView";

jest.mock("Platform", () => {
    const Platform = require.requireActual("Platform");
    Platform.OS = "android";
    return Platform;
});

it("renders PillView with titles", () => {
    const pillScenes = [
        {
            scene: (
                <View>
                    <Text>View 1</Text>
                </View>
            )
        },
        {
            scene: (
                <View>
                    <Text>View 2</Text>
                </View>
            )
        },
        {
            scene: (
                <View>
                    <Text>View 3</Text>
                </View>
            )
        }
    ];

    const pillHeaders = [
        { title: "View 1" },
        { title: "View 2" },
        { title: "View 3" }
    ];

    const tree = renderer
        .create(<PillView pillHeaders={pillHeaders} pillScenes={pillScenes} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders PillView with icons", () => {
    const pillScenes = [
        {
            scene: (
                <View>
                    <Text>View 1</Text>
                </View>
            )
        },
        {
            scene: (
                <View>
                    <Text>View 2</Text>
                </View>
            )
        },
        {
            scene: (
                <View>
                    <Text>View 3</Text>
                </View>
            )
        }
    ];

    const pillHeaders = [{ icon: "home" }, { icon: "card" }, { icon: "list" }];

    const tree = renderer
        .create(<PillView pillHeaders={pillHeaders} pillScenes={pillScenes} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});

it("renders PillView with titles and icons", () => {
    const pillScenes = [
        {
            scene: (
                <View>
                    <Text>View 1</Text>
                </View>
            )
        },
        {
            scene: (
                <View>
                    <Text>View 2</Text>
                </View>
            )
        },
        {
            scene: (
                <View>
                    <Text>View 3</Text>
                </View>
            )
        }
    ];

    const pillHeaders = [
        { title: "View 1", icon: "home" },
        { title: "View 2", icon: "card" },
        { title: "View 3", icon: "list" }
    ];

    const tree = renderer
        .create(<PillView pillHeaders={pillHeaders} pillScenes={pillScenes} />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
