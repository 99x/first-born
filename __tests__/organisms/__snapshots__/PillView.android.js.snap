// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PillView with icons 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 5,
        "marginHorizontal": 0,
        "marginTop": 35,
        "overflow": "hidden",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#486c86",
          "borderBottomWidth": 1.5,
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 10,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#486c86",
              "fontSize": 15,
            },
            Object {
              "marginRight": 5,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#adadad",
          "borderBottomWidth": 1.5,
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 10,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#adadad",
              "fontSize": 15,
            },
            Object {
              "marginRight": 5,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#adadad",
          "borderBottomWidth": 1.5,
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 10,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#adadad",
              "fontSize": 15,
            },
            Object {
              "marginRight": 5,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 16,
                  "fontWeight": "normal",
                  "textAlign": "left",
                },
              ]
            }
          >
            View 1
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`renders PillView with titles 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 5,
        "marginHorizontal": 0,
        "marginTop": 35,
        "overflow": "hidden",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#486c86",
          "borderBottomWidth": 1.5,
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 10,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#486c86",
              "fontSize": 13,
              "fontWeight": "normal",
              "textAlign": "left",
            },
          ]
        }
      >
        View 1
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#adadad",
          "borderBottomWidth": 1.5,
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 10,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#adadad",
              "fontSize": 13,
              "fontWeight": "normal",
              "textAlign": "left",
            },
          ]
        }
      >
        View 2
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#adadad",
          "borderBottomWidth": 1.5,
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 10,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#adadad",
              "fontSize": 13,
              "fontWeight": "normal",
              "textAlign": "left",
            },
          ]
        }
      >
        View 3
      </Text>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 16,
                  "fontWeight": "normal",
                  "textAlign": "left",
                },
              ]
            }
          >
            View 1
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`renders PillView with titles and icons 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
        "marginBottom": 5,
        "marginHorizontal": 0,
        "marginTop": 35,
        "overflow": "hidden",
      }
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#486c86",
          "borderBottomWidth": 1.5,
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 10,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#486c86",
              "fontSize": 15,
            },
            Object {
              "marginRight": 5,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#486c86",
              "fontSize": 13,
              "fontWeight": "normal",
              "textAlign": "left",
            },
          ]
        }
      >
        View 1
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#adadad",
          "borderBottomWidth": 1.5,
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 10,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#adadad",
              "fontSize": 15,
            },
            Object {
              "marginRight": 5,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#adadad",
              "fontSize": 13,
              "fontWeight": "normal",
              "textAlign": "left",
            },
          ]
        }
      >
        View 2
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "borderBottomColor": "#adadad",
          "borderBottomWidth": 1.5,
          "flex": 1,
          "flexDirection": "row",
          "justifyContent": "center",
          "opacity": 1,
          "paddingBottom": 10,
          "paddingVertical": 5,
        }
      }
    >
      <Text
        allowFontScaling={false}
        style={
          Array [
            Object {
              "color": "#adadad",
              "fontSize": 15,
            },
            Object {
              "marginRight": 5,
            },
            Object {
              "fontFamily": "Ionicons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
            Object {},
          ]
        }
      >
        
      </Text>
      <Text
        style={
          Array [
            Object {
              "color": "#adadad",
              "fontSize": 13,
              "fontWeight": "normal",
              "textAlign": "left",
            },
          ]
        }
      >
        View 3
      </Text>
    </View>
  </View>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 16,
                  "fontWeight": "normal",
                  "textAlign": "left",
                },
              ]
            }
          >
            View 1
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
